engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkRestManager.scala

```diff
@@ -27,38 +27,58 @@ class FlinkRestManager(config: FlinkConfig, modelData: BaseModelData, mainClassN
 
   private val modelJarProvider = new FlinkModelJarProvider(modelData.modelClassLoaderUrls)
 
-  private val client = new HttpFlinkClient(config)
+  private val client: FlinkClient = {
+    val httpClient = new HttpFlinkClient(config)
+
+    // It's not obvious that scenarioStateCaching enables Flink's Cache Client, it's just for simplify config
+    if (config.scenarioStateCaching.enabled) {
```

Comment:
It would be nice to duplicate this block - to have it done once and:
1. (preferred version) to parse config in once place and pass it in the more suitable form (option defined only when enabled = true) to both places
2. at least have the one method which will be used in both placese

Reply:
What about that?